src/journal-remote/journal-remote-util.c

```diff
@@ -0,0 +1,89 @@
+#include "extract-word.h"
+#include "journal-remote-util.h"
+#include "parse-util.h"
+
+void compression_args_clear(CompressionArgs *args) {
+        args->size = 0;
+        free(args->opts);
+}
+
+int config_parse_compression(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        CompressionArgs *args = ASSERT_PTR(data);
+        int r;
+        bool parse_level = ltype;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                compression_args_clear(args);
+                return 1;
+        }
+
+        for (const char *p = rvalue;;) {
+                _cleanup_free_ char *algorithm = NULL, *word = NULL;
+                int level = -1;
+
+                r = extract_first_word(&p, &word, NULL, 0);
+                if (r < 0)
+                        return log_syntax_parse_error(unit, filename, line, r, lvalue, rvalue);
+                if (r == 0)
+                        break;
+
+                const char *q = word;
+                r = extract_first_word(&q, &algorithm, ":", 0);
+                if (r < 0)
+                        return log_syntax_parse_error(unit, filename, line, r, lvalue, rvalue);
+
+                Compression c = compression_lowercase_from_string(algorithm);
+                if (c < 0 || !compression_supported(c)) {
+                        log_syntax(unit, LOG_WARNING, filename, line, c,
+                                   "Compression=%s is not supported on a system, ignoring", algorithm);
+                        continue;
+                }
+
+                if (!isempty(q) && parse_level) {
+                        r = safe_atoi(q, &level);
+                        if (r < 0) {
+                                log_syntax(unit, LOG_WARNING, filename, line, r,
+                                           "Compression level %s should be positive, ignoring", word);
+                                continue;
+                        }
+                }
+
+                bool found = false;
+                for (size_t i = 0; i < args->size; i++) {
+                        if (args->opts[i].algorithm == c) {
+                                found = true;
+                                if (parse_level)
+                                        args->opts[i].level = level;
+                                break;
+                        }
+                }
+
+                if (found) {
+                        continue;
+                }
+
+                if (!GREEDY_REALLOC(args->opts, (args->size + 1) * sizeof(CompressionOpts))) {
```

Comment:
`GREEDY_REALLOC()` takes the number of elements, not buffer size.
Also, please drop unnecessary `{}`.
Hence,
```c
if (!GREEDY_REALLOC(args->opts, args->size + 1))
        return log_oom();
```